x-pack/plugins/cases/server/client/cases/validators.ts

```diff
@@ -107,12 +107,26 @@ export const validateRequiredCustomFields = ({
     (customField) => customField.required
   );
 
+  if (!requiredCustomFields.length) {
+    return;
+  }
+
   const missingRequiredCustomFields = differenceWith(
     requiredCustomFields,
     requestCustomFields ?? [],
     (requiredVal, requestedVal) => requiredVal.key === requestedVal.key
   ).map((e) => e.key);
 
+  requiredCustomFields.forEach((requiredField) => {
+    const found = requestCustomFields?.find(
+      (requestField) => requestField.key === requiredField.key
+    );
+
+    if (found && found.value === null) {
```

Comment:
this will still allow empty value for text field, right?
In UI we don't allow empty string or string with only spaces for required field.

Reply:
ah, good point! I'll change it to `!found.value`

Reply:
it will fail for toggle with `false` value 😄  You will have to check `isEmpty(found.value) && typeof found.value === 'string'`

Reply:
I have to think before writing :D

Reply:
Ah but actually it is more complex than that.

The text type is supposed to be an array of strings so `typeof found.value === 'string'` will never happen.

We need to check for `null`, `[]` or `['']`.

Reply:
Can't we do it on the schema level?

Reply:
Pushed changes.

To sum up:
- validator checks for null value
- schema checks for empty array on `text` custom fields
- schema checks for empty string in `value` array for `text` custom fields